Regression for JAXB annotations module (for 2.16.0-SNAPSHOT) due to `jackson-databind` enum changes

Looks like JAXB annotations module has 1 new failing unit test in `2.16` branch:

```
[ERROR] Failures: 
[ERROR]   TestJaxbAnnotationIntrospector.testSerializeDeserializeWithJaxbAnnotations:266 expected:<V[alue One]> but was:<V[ALUE1]>
[INFO] 
[ERROR] Tests run: 81, Failures: 1, Errors: 0, Skipped: 0
```

which I think is due to recent changes to Enum lookup wrt annotations.

--- a/jakarta-xmlbind/src/main/java/com/fasterxml/jackson/module/jakarta/xmlbind/JakartaXmlBindAnnotationIntrospector.java
+++ b/jakarta-xmlbind/src/main/java/com/fasterxml/jackson/module/jakarta/xmlbind/JakartaXmlBindAnnotationIntrospector.java
@@ -951,6 +951,36 @@ public class JakartaXmlBindAnnotationIntrospector
         return names;
     }
 
+    /**
+     * @see JacksonAnnotationIntrospector#findEnumValues(MapperConfig, AnnotatedClass, Enum[], String[]) 
+     * @since 2.16
+     */
+    @Override
+    public String[] findEnumValues(MapperConfig<?> config, AnnotatedClass annotatedClass,
+                                   Enum<?>[] enumValues, String[] names)
+    {
+        Map<String, String> enumToPropertyMap = new LinkedHashMap<String, String>();
+        for (AnnotatedField field : annotatedClass.fields()) {
+            XmlEnumValue property = field.getAnnotation(XmlEnumValue.class);
+            if (property != null) {
+                String propValue = property.value();
+                if (propValue != null && !propValue.isEmpty()) {
+                    enumToPropertyMap.put(field.getName(), propValue);
+                }
+            }
+        }
+
+        // and then stitch them together if and as necessary
+        for (int i = 0, end = enumValues.length; i < end; ++i) {
+            String defName = enumValues[i].name();
+            String explValue = enumToPropertyMap.get(defName);
+            if (explValue != null) {
+                names[i] = explValue;
+            }
+        }
+        return names;
+    }
+    
     /*
     /**********************************************************************
     /* Deserialization: general annotations

--- a/jaxb/src/main/java/com/fasterxml/jackson/module/jaxb/JaxbAnnotationIntrospector.java
+++ b/jaxb/src/main/java/com/fasterxml/jackson/module/jaxb/JaxbAnnotationIntrospector.java
@@ -969,6 +969,36 @@ public class JaxbAnnotationIntrospector
         return names;
     }
 
+    /**
+     * @see JacksonAnnotationIntrospector#findEnumValues(MapperConfig, AnnotatedClass, Enum[], String[])
+     * @since 2.16
+     */
+    @Override
+    public String[] findEnumValues(MapperConfig<?> config, AnnotatedClass annotatedClass,
+                                   Enum<?>[] enumValues, String[] names)
+    {
+        Map<String, String> enumToPropertyMap = new LinkedHashMap<String, String>();
+        for (AnnotatedField field : annotatedClass.fields()) {
+            XmlEnumValue property = field.getAnnotation(XmlEnumValue.class);
+            if (property != null) {
+                String propValue = property.value();
+                if (propValue != null && !propValue.isEmpty()) {
+                    enumToPropertyMap.put(field.getName(), propValue);
+                }
+            }
+        }
+
+        // and then stitch them together if and as necessary
+        for (int i = 0, end = enumValues.length; i < end; ++i) {
+            String defName = enumValues[i].name();
+            String explValue = enumToPropertyMap.get(defName);
+            if (explValue != null) {
+                names[i] = explValue;
+            }
+        }
+        return names;
+    }
+
     /*
     /**********************************************************
     /* Deserialization: general annotations

